pkg/install/process.go

```diff
@@ -16,135 +16,67 @@ limitations under the License.
 package install
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"os/exec"
-	"runtime"
-	"strings"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
 
-	"github.com/k0sproject/k0s/internal/util"
-	"github.com/mitchellh/go-ps"
-	"gopkg.in/yaml.v2"
+	config "github.com/k0sproject/k0s/pkg/apis/v1beta1"
+	"github.com/k0sproject/k0s/pkg/constant"
 )
 
 type K0sStatus struct {
-	Version  string
-	Pid      int
-	PPid     int
-	Role     string
-	SysInit  string
-	StubFile string
-	Output   string
+	Version       string
+	Pid           int
+	PPid          int
+	Role          string
+	SysInit       string
+	StubFile      string
+	Output        string
+	Workloads     bool
+	Args          []string
+	ClusterConfig *config.ClusterConfig
+	K0sVars       constant.CfgVars
 }
 
-func GetPid() (status *K0sStatus, err error) {
-	pid, ppid, err := getProcessID()
-	if err == nil && pid != nil {
-		status = &K0sStatus{
-			Pid:  *pid,
-			PPid: *ppid,
-		}
-		return status, nil
-	}
-	return &K0sStatus{}, nil
-}
+func GetStatusInfo(runDir string) (status []K0sStatus, err error) {
+	roles := []string{"controller", "worker"}
+	status = []K0sStatus{}
 
-func getProcessID() (pid *int, ppid *int, err error) {
-	processList, err := ps.Processes()
-	if err != nil {
-		return nil, nil, err
-	}
-
-	for _, p := range processList {
-		if p.Executable() == "k0s" && hasChildren(p.Pid(), processList) {
-			pid := p.Pid()
-			ppid := p.PPid()
-			return &pid, &ppid, nil
+	for _, role := range roles {
+		address := filepath.Join(runDir, fmt.Sprintf("%s.sock", role))
```

Comment:
There's probably something going wrong if a rundir has data for multiple instances. A program should usually fail to start if the rundir indicates there's an instance already active (a pidfile exists in the rundir and the process listed in it is alive). 

I feel like there's all this trickery going on just to avoid fixing the constant rundir (for root, for regular users it's under data-dir/run).

The rationale for creating a status socket was to confidently get data from a specific instance.

Reply:
Let's open another PR for configurable runDir

Reply:
#991